packages/health/ios/Classes/SwiftHealthPlugin+Data.swift

```diff
@@ -0,0 +1,245 @@
+import HealthKit
+
+extension SwiftHealthPlugin {
+
+  func writeData(call: FlutterMethodCall, result: @escaping FlutterResult) {
+    guard let arguments = call.arguments as? NSDictionary,
+          let value = (arguments["value"] as? Double),
+          let type = (arguments["dataTypeKey"] as? String),
+          let startDate = (arguments["startTime"] as? NSNumber),
+          let endDate = (arguments["endTime"] as? NSNumber),
+          let overwrite = (arguments["overwrite"] as? Bool) else {
+      DispatchQueue.main.async {
+        result(PluginError(message: "Invalid Arguments"))
+      }
+      return
+    }
+    guard let dataType = HealthDataTypes(rawValue: type) else {
+      DispatchQueue.main.async {
+        result(PluginError(message: "Unrecognized dataTypeKey \(type)"))
+      }
+      return
+    }
+
+    logger.debug("\(#function) \(dataType.rawValue) \(value)")
+
+    let dateFrom = Date(timeIntervalSince1970: startDate.doubleValue / 1000)
+    let dateTo = Date(timeIntervalSince1970: endDate.doubleValue / 1000)
+
+    let sample: HKObject? = if let unit = dataType.unit, let quantityType = dataType.quantityType {
+      HKQuantitySample(
+        type: quantityType,
+        quantity: HKQuantity(unit: unit, doubleValue: value),
+        start: dateFrom,
+        end: dateTo
+      )
+    } else if let categoryType = dataType.categoryType {
+      HKCategorySample(
+        type: categoryType,
+        value: Int(value),
+        start: dateFrom,
+        end: dateTo
+      )
+    } else {
+      nil
+    }
+    guard let sample else {
+      DispatchQueue.main.async {
+        result(PluginError(message: "Failed to create HKObject for \(type)"))
+      }
+      return
+    }
+
+    let saveOperation = {
+      healthStore.save(sample) { (success, error) in
+        if let error {
+          logger.error("\(#function) saving \(type) failed: \(error.localizedDescription)")
+        }
+        DispatchQueue.main.async {
+          result(success)
+        }
+      }
+    }
+
+    if overwrite {
+      guard let categoryType = dataType.categoryType else {
+        logger.error("\(#function) deleting \(type) failed: HKCategoryType could not be resolved")
+        saveOperation()
+        return
+      }
+      healthStore.deleteObjects(
+        of: categoryType,
+        predicate: HKQuery.predicateForSamples(
+          withStart: dateFrom,
+          end: dateTo,
+          options: []
+        )
+      ) { (_, _, error) in
+        if let error {
+          logger.error("\(#function) deleting \(type) failed: \(error.localizedDescription)")
+        }
+        saveOperation()
+      }
+    } else {
+      saveOperation()
+    }
```

Comment:
This behavior is new. For overwrites, this now performs the save after the delete